docs/RFCS/20200401_alter_column_type.md

```diff
@@ -0,0 +1,87 @@
+
+Interesting page on ALTER COLUMN POSITION idea for Postgres.
+https://wiki.postgresql.org/wiki/Alter_column_position
+
+This blurb mentions using separate logical and physical identifiers for the column.
+```
+Adding alter column syntax into postgres
+Since the above methods have a number of issues, it has often been expressed that we would like to add capabilities for postgres to allow reordering of columns. The flip side of this is that it would also be desirable for postgres to automatically order columns physically for optimum layout, regardless of the logical order they are given in. The current problem with implementing this lies in that currently postgres uses the same identifiers for both the logical and physical position within a table. The current hot plan for solving this would be to change the system to reference three identifiers... a permanent identifier for the column, as well as a separate logical and physical identifier. This would allow places that need to deal specifically with column order at the logical level (ie. select *) to reference the logical number, while places that interact with disk system can access the physical number, and all other places just use the column's permanent id.
+
+The above plan is taken from the following email http://archives.postgresql.org/pgsql-hackers/2006-12/msg00983.php.
+The TODO list references this email, http://archives.postgresql.org/pgsql-hackers/2006-12/msg00782.php, which is a fun read if you want more back story.
+The other bit is that we would also need to determine syntax for how users would interact with this new functionality. The most common implementations typically use a set of BEFORE/AFTER keywords, followed by an existing column name.
+```
+## USING EXPRESSION for conversion
+If we’re planning to support USING EXPRESSION for altering a column data type, we run into the issue where if one node can see the old version of a TableDescriptor and if one node has the new version, if insert happens on the new version into the column with the new data type, and a read happens from the node with the old TableDescriptor.
+
+It’s alright if we’re able to cast back and forth between the two types, however if we used an expression to alter the type, it may be impossible to convert back to the old type (need an inverse function.)
+
+For example, if we're converting a column from int to bool and using the expression f(x) = true if x > 0 and false otherwise, how do we get an inverse for this function?
+
+# Other considerations
```

Comment:
can you add a section about what should happen if there is an invalid cast from one to another?
e.g. let's say we are converting from string to timestamp. one of the timestamp string values is invalid. how do we abort?

Reply:
Added a section.